Merge legacy and mining /blocks components and APIs

nymkappa opened the pull request:
Fixes https://github.com/mempool/mempool/issues/1692

This PR completely gets rid of the legacy `/blocks` page (currently seen when the mining dashboard is disabled in the frontend config, or for non Bitcoin networks) and its related `/blocks` node backend API.

There is now one single `blocks-list` component which properly display what is available according to the currently running environment.

The legacy `/blocks` API has been deleted, and the `/blocks-extras` API has been renamed as `/blocks`.

softsimon commented:
The `mining-dashboard.component.html` and `dashboard.component.html` still do routerLink to `/mining/blocks`

My latest block widget is suddenly very long vertically

<img width="579" alt="Screen Shot 2022-05-20 at 22 19 42" src="https://user-images.githubusercontent.com/8561090/169589276-ed04531a-9709-4a56-b01b-841489f9d8ae.png">

softsimon commented on frontend/src/app/components/blocks-list/blocks-list.component.html:
```diff
@@ -1,29 +1,30 @@
-        <th class="fees text-right" i18n="latest-blocks.fees" *ngIf="!widget">Fees</th>
-        <th class="txs text-right" i18n="dashboard.txs" [class]="widget ? 'widget' : ''">TXs</th>
-        <th class="size" i18n="latest-blocks.size" *ngIf="!widget">Size</th>
+        <th class="height text-left" [class]="widget ? 'widget' : ''" [class]="indexingAvailable ? '' : 'legacy'" i18n="latest-blocks.height">Height</th>
```
Maybe it's not possible to combine multiple `[class]` without conflict.

Instead consider:

`[ngClass]="{'widget: widget, 'legacy': !indexingAvailable}"`

nymkappa replied:
That fixed it, thanks!

nymkappa commented:
* Added redirection from `/mining/blocks` to `/blocks` on bitcoin mainnet/tesnet/signet in routing module
* Fixed [class] conflict using @softsimon suggestion
* Update outdated [router] link so they directly point to `/blocks`

nymkappa commented:
Rebased on master

softsimon commented:
Seems to break on Liquid.

`liquid/api/v1/blocks`

`Cannot read properties of undefined (reading 'coinbaseRaw')`

And skeleton loader seems not right with the columns
<img width="1325" alt="Screen Shot 2022-05-23 at 13 35 07" src="https://user-images.githubusercontent.com/8561090/169790255-971eb81b-485d-4309-a5a4-00b939fc11db.png">

wiz reviewed (approved):
Tested ACK @ v2.4.0-dev [[aac9dda9](https://github.com/mempool/mempool/commit/aac9dda9)]

Tried all networks, seems fine
